lib/kazoo/partition.rb

```diff
@@ -60,6 +60,19 @@ def hash
       [topic, id].hash
     end
 
+    def wait_for_leader
+      current_leader = nil
+      while current_leader.nil?
+        current_leader = begin
+          leader
+        rescue Kazoo::Error
+          nil
+        end
+
+        sleep(0.1) if current_leader.nil?
+      end
+    end
+
```

Comment:
Potentially this could be done using Zookeeper watches, but this is very complicated because of all the nodes that need to be created in Zookeeper as part of the creation process.